modules/navier_stokes/src/physics/WCNSFVFluidHeatTransferPhysics.C

```diff
@@ -403,16 +482,67 @@ WCNSFVFluidHeatTransferPhysics::addEnergyWallBC()
   }
 }
 
+void
+WCNSFVFluidHeatTransferPhysics::addMaterials()
+{
+  if (!_has_energy_equation)
```

Comment:
hopefully we can remove this boolean soon?

Reply:
yeah at some point. I would have to transition the rest of the inputs.